### PR TITLE
Upgrade Azure Go SDK to stable version

### LINKED ISSUE #63056: Upgrade Azure Go SDK to stable version
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
/sig azure

**What happened**:

Kubernetes is using a beta version of Azure Go SDK now. If there are bugs in them, it's hard to upgrade because Azure Go SDK won't release new patches for pre-released SDK versions. We should upgrade Go SDK to stable version (e.g.  v14.6.0)

Refer https://github.com/kubernetes/kubernetes/pull/62249

Refer https://github.com/Azure/azure-sdk-for-go/issues/1586

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/assign

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Kubernetes is using a beta version of Azure Go SDK now. If there are bugs in them, it's hard to upgrade because Azure Go SDK won't release new patches for pre-released SDK versions. We should upgrade Go SDK to stable version (e.g. v14.6.0)

Refer #62249

Refer Azure/azure-sdk-for-go#1586

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63056

**Special notes for your reviewer**:

This PR includes changes in #61972, but with a newer go-autorest version.

**Release note**:

```release-note
Upgrade Azure Go SDK to stable version (v14.6.0)
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
@karataliu PTAL, this is a big change

File: pkg/cloudprovider/providers/azure/azure_backoff.go [hunk 1]
```diff
@@ -89,30 +89,6 @@ func (az *Cloud) VirtualMachineClientListWithRetry() ([]compute.VirtualMachine,
 		return nil, err
 	}
 
-	appendResults := (result.Value != nil && len(*result.Value) > 0)
```
[the remark this answers is not in the document]
Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_backoff.go]: VirtualMachinesClient.List() already returns []compute.VirtualMachine

**Comment by feiskyer** [comment]
ping @brendandburns

**Comment by feiskyer** [comment]
Rebased. @brendandburns PTAL

**Comment by feiskyer** [comment]
@brendandburns Thanks

**Pull request merged**